waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -863,7 +864,9 @@
                                            service-id->failed-instances-transient-store
                                            service-id->password-fn
                                            service-id->service-description-fn
-                                           watch-state)]
-      (reset! daemon-state {:pod-watch-thread (start-pods-watch! scheduler)
-                            :rs-watch-thread (start-replicasets-watch! scheduler)})
+                                           watch-state)
+          pod-watch-thread (start-pods-watch! scheduler)
+          rs-watch-thread (start-replicasets-watch! scheduler)]
+      (reset! daemon-state {:pod-watch-cancel #(.stop pod-watch-thread)
+                            :rs-watch-cancel #(.stop rs-watch-thread)})
```

Comment:
I'm aware of this, and there's a reason I chose to use `.stop` despite the deprecation. Unlike the core.async threads we have elsewhere, these threads use non-interruptible blocking i/o and have no guarantee to ever see the cancellation signal that you're suggesting.

Reply:
To clarify further, *cheshire* is using non-interruptible blocking i/o. Unless we completely re-write the streaming JSON parsing routines, an unclean `Thread/stop` is the only way to kill these.

However, since we never kill the threads outside of the unit tests (we never actually call the functions created here—the unit tests call `start-k8s-watch!` directly), I we could simply drop this entire `daemon-state` atom. If we want something like this again later, we can always add it back in.

I'll just drop the references to `Thread/stop` in the main code for now. We'll still be using `.stop` in the unit tests, but at least we won't have it in the main code, so it won't look as hacky.

Reply:
I'm happy to see no `Thread/stop` in the main code. I'm aware that cheshire is non-interruptible. The proposed cooperative strategy with the atom is to perform the termination when control comes back to the function when cheshire has returned a single object.

Reply:
The problem is that the read can block indefinitely (e.g., it might be an hour before the watch thread sees an update), so we can't depend on cooperative termination.